### PR TITLE
feat: accept `Uint8Array`s as inputs for `Vec`s, add docs for bytecode inputs

### LINKED ISSUE #1950: Add a helper for the conversion: bytecode file contents -> `Vec<u8>`
We need a helper function to read bytecode from a file and convert that to a bytearray that users can pass into our `sway-libs` functions related to bytecode: https://github.com/FuelLabs/sway-libs/tree/master/libs/bytecode

This conversion is required because the bytecode from a file is a `string`, but the type that these Sway functions accept for the bytecode is a `Vec<u8>` - which in JavaScript-land is a `Vec<BigNumberish>`.

This helper would convert the string to the proper type and ensure all the data is correctly packed.

## Possible API
```ts
const bytecodeFromFile = readFile(pathToBytecode); // string

const bytecodeVec = generateBytecodeVecFromString(bytecodeFromFile); // our helper -> returns `Vec<BigNumberish>`

await contract.functions.verify_bytecode(bytecodeVec);
```

> _Incoming from feedback from the development work being done on the Fuelnaut project: https://github.com/FuelLabs/docs-hub/pull/203_

The Fuelnaut project is blocked for now until we figure this out.

### PR DESCRIPTION
**Pull request by Dhaiwat10**
## Summary

Before this PR, the `VecCoder` wasn't accepting input arrays of type `Uint8Array`. In some cases, like the `compute_bytecode_root` function from the [bytecode Sway Lib](https://github.com/FuelLabs/sway-libs/tree/master/libs/src/bytecode), you need to pass in a `Uint8Array` to the Sway function.

I have tweaked the validation inside the `VecCoder` for the inputs, and also added some documentation around how to pass in bytecode to Sway functions.

You can pass in bytecode to any Sway function like so now:

```ts
import { arrayify } from 'fuels';

const { logs } = await contract.functions
  .compute_bytecode_root(arrayify(bytecodeFromFile))
  .call();
```

Closes #1950 

This PR will unblock the Fuelnaut project, too.

### CODE REVIEW

**Review by danielbate — COMMENTED** [review]
Hmm IMO a `Vec<u8>` representing a byte array should actually be taking `Bytes` as the input? Which would then be picked up by the `ByteCoder` which accepts `UInt8Array`. Is this a sway built in function which accepts `Vec<u8>` when expecting bytes?

**Comment by Dhaiwat10** [comment]
@danielbate yes that is the case. All the functions in this Sway library (which is officially maintained and supported by us) accept bytecode as `Vec<u8>`

File: apps/docs-snippets/test/fixtures/forc-projects/bytecode-input/.gitignore
Comment (danielbate) [inline apps/docs-snippets/test/fixtures/forc-projects/bytecode-input/.gitignore:1]: This file is redundant.

File: apps/docs-snippets/test/fixtures/forc-projects/bytecode-input/Forc.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[project]
+authors = ["Fuel Labs <contact@fuel.sh>"]
+license = "Apache-2.0"
+name = "bytecode-input"
+
+[dependencies]
+bytecode = { git = "https://github.com/FuelLabs/sway-libs", tag = "v0.19.0" }
```
Comment (danielbate) [inline apps/docs-snippets/test/fixtures/forc-projects/bytecode-input/Forc.toml:7]: Why do we need to pin for an older version? My concern is that we tend to keep all our examples fixed to a specific verison.

File: apps/docs/src/guide/types/vectors.md [hunk 2]
```diff
@@ -20,6 +20,18 @@ The code snippet below demonstrates how to call this Sway contract method, which
...
 
+## Working with Bytecode in the SDK
+
+Some Sway functions require you to pass in bytecode to the function. The type of the bytecode parameter is usually `Vec<u8>`.
+
+Take the `compute_bytecode_root` function from the [`bytecode` Sway library](https://github.com/FuelLabs/sway-libs/tree/master/libs/src/bytecode.sw), for example.
+
+<<< @/../../docs-snippets/test/fixtures/forc-projects/bytecode-input/src/main.sw#vector-bytecode-input-sway{ts:line-numbers}
+
+To pass in some bytecode to this function, you can make use of the `arrayify` function to convert the bytecode file contents into a `Vec<u8>` type and pass it to the function like this:
```
Comment (danielbate) [inline apps/docs/src/guide/types/vectors.md:31]:
```suggestion
To pass bytecode to this function, you can make use of the `arrayify` function to convert the bytecode file contents into a `UInt8Array`, the TS compatible type for Sway's `Vec<u8>` type and pass it the function like so:
```

Just amended to show the TS -> Sway flow.

File: apps/docs-snippets/test/fixtures/forc-projects/bytecode-input/Forc.toml (hunk 1 above)
Comment (arboleya) [reply apps/docs-snippets/test/fixtures/forc-projects/bytecode-input/Forc.toml:7]: Absolutely, good catch! I second this concern. ☝️

File: apps/docs-snippets/test/fixtures/forc-projects/bytecode-input/.gitignore
Comment (Dhaiwat10) [reply apps/docs-snippets/test/fixtures/forc-projects/bytecode-input/.gitignore:1]: https://github.com/FuelLabs/fuels-ts/pull/2018/commits/fe911ee3de1a663e0d8b520494d2ed6072b794c6

File: apps/docs-snippets/src/guide/types/vector.test.ts [hunk 3]
```diff
@@ -52,4 +54,28 @@ describe(__filename, () => {
...
+      DocSnippetProjectsEnum.BYTECODE_INPUT
+    );
+    const bytecodePath = join(
+      __dirname,
+      '../../../test/fixtures/forc-projects/bytecode-input/out/release/bytecode-input.bin'
+    );
+
+    // #region vector-bytecode-input-ts
+    // #import { arrayify };
+    // #context import { readFile } from 'fs/promises';
```
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply apps/docs-snippets/src/guide/types/vector.test.ts:69]: wouldnt that import `readFile` from fuels? instead of `fs/promises`?

**Review by danielbate — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**